packages/flutter/lib/src/painting/rounded_rectangle_border.dart

```diff
@@ -0,0 +1,298 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+
+import 'basic_types.dart';
+import 'border_radius.dart';
+import 'borders.dart';
+import 'circle_border.dart';
+import 'edge_insets.dart';
+
+/// A rectangular border with rounded corners.
+///
+/// Typically used with [ShapeDecoration] to draw a box with a rounded
+/// rectangle.
+///
+/// This shape can interpolate to and from [CircleBorder].
+///
+/// See also:
+///
+///  * [BorderSide], which is used to describe each side of the box.
+///  * [Border], which, when used with [BoxDecoration], can also
+///    describe a rounded rectangle.
+class RoundedRectangleBorder extends ShapeBorder {
+  /// Creates a rounded rectangle border.
+  ///
+  /// The arguments must not be null.
```

Comment:
Maybe "None of the arguments may be null."?

Reply:
I try and avoid negation and "may". People get really confused around it.